docs/io/index_unofficial.rst

```diff
@@ -0,0 +1,283 @@
+
+* **Your reader can be entirely programmed in Python**, in which case your
+  reader does not depend on any external packages, and you can include that
+  code in your reader's :ref:`Low-Level Reader-Specific Modules`.
+* **Your reader exists as an external Python module**, in which case it could
+  be made available on conda for easy (optional) installion.
+* **You have a C/C++ inplementation of your reader in a simple script**,
+  therefore you could provide an interface to that functionality by using
+  "ctypes". The C/C++ scripts can exist inside your reader module, and are
+  compiled as part of the ctapipe installation. See
+  `ctapipe.utils.neighbour_sum` as an example of how to interface Python with
+  C code, and remember to add your extension to ext_modules in setup.py.
+* **Your reader exists as an external C/C++ library**, therefore software such
+  as SWIG can be used to provide a Python wrapper to the library,
+  automatically creating interfaces to the functions in the library.
+
+
+EventFileReaders (`ctapipe.io.unofficial.eventfilereader`)
+==========================================================
+
+This class provides the high-level interface for reading data from each data
+format. Read about the purpose of EventFileReaders at
+:ref:`EventFileReaderClasses`.
+
+Location for every unofficial `ctapipe.io.eventfilereader.EventFileReader`.
+
+Each reader is required to import their
+:ref:`Low-Level Reader-Specific Modules` inside their `__init__` method,
+therefore bypassing the need for the external reader software to be installed
+unless someone has chosen to use this specific
+`ctapipe.io.eventfilereader.EventFileReader`, in which case a
+`ModuleNotFoundError` will be raised if the external software has not been
+found.
+
+The reader must also implement a fast method for
+`ctapipe.io.eventfilereader.EventFileReader.check_file_compatibility`. This
+method attempts to find a reader which *may* be compatible with the input
+file, and can be as simple as looking for a unique file extension in the file
+path or attempting to find a header keyword in the file. It is strongly
+advised that this method does not depend on the external reader software, as
+this would introduce a dependency when choosing the correct reader in the
+`ctapipe.io.eventfilereader.EventFileReaderFactory`.
+
+The currently existing EventFileReaders,
+`ctapipe.io.eventfilereader.HessioFileReader` and
+`ctapipe.io.unofficial.eventfilereader.TargetioFileReader` can be used as
+templates to create a new EventFileReader for a different data format.
+
+
+Low-Level Reader-Specific Modules
+===========================
+
+A seperate module should be created for the low-level method used for filling
+the event containers for a specific data format. The naming convention for
+this module is "\*io.py", where "\*" is the short-name for the data format.
+For example: `ctapipe.io.hessio` and `ctapipe.io.unofficial.targetio.targetio`.
```

Comment:
Really? We put sub-modules into the `io` module and invent a rule, that sub-module names have to **end on `io`** as well?

Reply:
Its just the convention we have been using so far.

Reply:
I don't think it was a convention to call the submodule `*io`, I think it was just an chance that all readers until now are for formats that end on `io` like hessio and targetio.

Reply:
And i also do not like to have io at least two times in there if it is not necessary.